staging/src/k8s.io/kubectl/pkg/cmd/debug/profile_options.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const ProfileLegacy = "legacy"
+
+// applierFunc is a function that applies a profile to a container in the pod.
+type applierFunc func(pod *corev1.Pod, containerName string, target runtime.Object) error
+
+type ProfileOptions struct {
+	applier applierFunc
+}
+
+// NewProfile returns a new Options for the given profile name.
+func NewProfile(profile string) (*ProfileOptions, error) {
```

Comment:
With the above suggestion I'd rename this to `ToProfileApplier` or `NewProfileApplier`